unittests/test_dependencies.py

```diff
@@ -174,10 +199,23 @@ def test_dependecies_when_functions():
     }
     assert len(deps) == 4
 
-    when = udeps.part_is('p0')
+    how = udeps.by_xcase
+    deps = {(t0, t1)
+            for t0 in t0_cases
+            for t1 in t1_cases
+            if how(t0, t1)}
+    assert deps == {
+        (t0, t1)
+        for t0 in t0_cases
+        for t1 in t1_cases
+        if (t0[0] != t1[0] or t0[1] != t1[1])
+    }
+    assert len(deps) == 32
+
+    how = udeps.part_is('p0')
```

Comment:
From this point onward, everything should go inside a different test function: `test_dependecies_how_functions_undoc`

Reply:
Just to be clear, `env_is`, `part_is`, `source`, `dest`, `all`, `any` will be all undocumented, until we decide about their future.